builder/role_image.go

```diff
@@ -300,14 +277,42 @@ func (r *RoleImageBuilder) generateDockerfile(instanceGroup *model.InstanceGroup
 		return err
 	}
 
-	dockerfileTemplate := template.New("Dockerfile-role")
+	// In a role move the packages from /var/vcap/packages-src to
+	// /var/vcap/packages.  This replaces the original form of
+	// symlinking them into place.  The change is necessary
+	// because the symlinks will not resolve inside bpm containers.
+
+	packageSet := map[string]string{}
+	moves := []string{}
```

Comment:
I don't like the name `moves`; these are really `commands` of any kind.

I can't make up my mind, but I think I might even prefer to have the `RUN` prefix in the template, and just add regular shell commands to a `runCommands` array that you expand with

```
{{ range .runCommands }}
RUN {{ . }}
{{ end }}
```

Reply:
Made this change, plus a few other fixes: Typo in the paths, and forgot to create destination directory.
Ref commit 6a73892c23d0eac50861ac6c031012d338d90a80